gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java

```diff
@@ -68,7 +68,7 @@ public TimeAwareRecursiveCopyableDataset(FileSystem fs, Path rootPath, Propertie
     this.datePattern = properties.getProperty(DATE_PATTERN_KEY);
 
     this.currentTime = properties.containsKey(DATE_PATTERN_TIMEZONE_KEY) ? LocalDateTime.now(
-        DateTimeZone.forID(DATE_PATTERN_TIMEZONE_KEY))
+        DateTimeZone.forID(properties.getProperty(DATE_PATTERN_TIMEZONE_KEY)))
         : LocalDateTime.now(DateTimeZone.forID(DEFAULT_DATE_PATTERN_TIMEZONE));
```

Comment:
I find it written most succinctly in - https://github.com/apache/gobblin/blob/d9ae5353c74fdcd385835fca9b586b3fdb90971b/gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/UnixTimestampRecursiveCopyableDataset.java#L76

BTW, these two could also be DRYed up:
https://github.com/apache/gobblin/blob/d9ae5353c74fdcd385835fca9b586b3fdb90971b/gobblin-data-management/src/main/java/org/apache/gobblin/data/management/version/finder/DatePartitionHiveVersionFinder.java#L69
https://github.com/apache/gobblin/blob/d9ae5353c74fdcd385835fca9b586b3fdb90971b/gobblin-data-management/src/main/java/org/apache/gobblin/data/management/version/finder/DateTimeDatasetVersionFinder.java#L86

Reply:
I'll follow the UnixTimestampRecursiveCopyableDataset pattern here, thanks for the heads up :).

For the other 2 maybe we can find a way to aggregate lookback and timezone configurations across Gobblin jobs so I'll leave them for now. Seems like a recurring problem in Gobblin where you have 2 configurations with the same semantics but under different configurations due to being part of different workflows.

Reply:
that's a worthwhile pursuit.... but complicated because our installed/legacy user base may have come to depend on the old names.  in general they can't be removed... only given an alias.  this seems in the spirit of attacking that proliferation:
https://github.com/apache/gobblin/blob/d9ae5353c74fdcd385835fca9b586b3fdb90971b/gobblin-utility/src/main/java/org/apache/gobblin/util/deprecation/DeprecationUtils.java#L43

but, yes, I agree on the benefit of aligning under a common name for the 'same' concept in use across multiple classes.  at this point, I'm don't have an opinion on which name deserves to 'win' and be enshrined as the 'standard property name'.

(all that... and... my original suggestion was just to DRY out the code, eliminating the duplicated calls, not specifically to standardize on a common property name.)